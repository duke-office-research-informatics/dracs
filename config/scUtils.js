import React from "react";
import { ThemeProvider } from "styled-components";
import { mount, shallow } from "enzyme";
import { theme as defaultTheme } from "../lib/dracs.es.js";

export const shallowWithTheme = (children, customTheme) => {
  const theme = customTheme || defaultTheme;
  return shallow(children, { context: theme });
};

/* eslint-disable */
export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}

//FOR ACCESSING PORTALS VIA ENZYME -- import portal from dracs
// https://github.com/airbnb/enzyme/issues/1202#issuecomment-351639783
// function mountPortal(el){
//   const wrapper =  mountWithTheme(el);
//   wrapper.setState({ visible: true});
//   const portalInstance = wrapper.find(Portal).instance();
//   return mountWithTheme(portalInstance.props.children);
// }

//FOR ACCESSING EVENT LISTENERS ATTACHED TO DOCUMENT OR DOCUMENT.BODY
//https://github.com/airbnb/enzyme/issues/426#issuecomment-296304015
// const map = {};
// document.body.addEventListener = jest.fn((event, cb) => {
//   map[event] = cb;
// });
