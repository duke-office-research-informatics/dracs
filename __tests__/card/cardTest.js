import React from "react";
import { mountWithTheme } from "../../config/scUtils.js";
import { axe } from "jest-axe";
import { theme, Card, IconMenu } from "../../lib/dracs.es.js";

const mockFn = jest.fn();

describe("Card", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mountWithTheme(<Card />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("does not return any a11y violations", async () => {
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  it("sets background color via props", () => {
    wrapper.setProps({ bgColor: theme.colors.dashBG });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("card__CardWrapper")).toHaveStyleRule(
      "background-color",
      theme.colors.dashBG
    );
  });

  it("does not apply a border when `border` set to false", () => {
    wrapper.setProps({ border: false });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("card__CardWrapper")).not.toHaveStyleRule(
      "border",
      `solid 1px ${theme.colors.border}`
    );
  });

  it("attaches children", () => {
    const card = mountWithTheme(
      <Card>
        <IconMenu />
      </Card>
    );
    expect(card).toMatchSnapshot();
    expect(card.find(IconMenu)).toBeDefined();
  });

  it("attaches a className via props", () => {
    wrapper.setProps({ className: "testClass" });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("card__CardWrapper").hasClass("testClass")).toBe(true);
  });

  it("sets height via props", () => {
    wrapper.setProps({ height: "300px" });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("card__CardWrapper")).toHaveStyleRule(
      "height",
      "300px"
    );
  });

  it("sets margin via props", () => {
    wrapper.setProps({ margin: "20px" });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("card__CardWrapper")).toHaveStyleRule("margin", "20px");
  });

  it("sets minHeight via props", () => {
    wrapper.setProps({ minHeight: "200px" });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("card__CardWrapper")).toHaveStyleRule(
      "min-height",
      "200px"
    );
  });

  it("sets minWidth via props", () => {
    wrapper.setProps({ minWidth: "200px" });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("card__CardWrapper")).toHaveStyleRule(
      "min-width",
      "200px"
    );
  });

  it("sets an onClick handler that is called on click", () => {
    wrapper.setProps({ onClick: mockFn });
    expect(wrapper).toMatchSnapshot();
    wrapper.find("card__CardWrapper").simulate("click");
    expect(mockFn).toHaveBeenCalled();
  });

  it("sets padding via props", () => {
    wrapper.setProps({ padding: "8px" });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("card__CardWrapper")).toHaveStyleRule("padding", "8px");
  });

  it("adds a box-shadow when `raised` is set to true", () => {
    wrapper.setProps({ raised: true });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("card__CardWrapper")).toHaveStyleRule(
      "box-shadow",
      "0 1px 3px 0 rgba(0,0,0,0.25)"
    );
  });

  it("sets a style object via props", () => {
    wrapper.setProps({ style: { flexDirection: "row" } });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("card__CardWrapper").props().style).toHaveProperty(
      "flexDirection",
      "row"
    );
  });

  it("sets a width via props", () => {
    wrapper.setProps({ width: "100px" });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("card__CardWrapper")).toHaveStyleRule("width", "100px");
  });

  it("sets an html id attr via props", () => {
    wrapper.setProps({ id: "testID" });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("card__CardWrapper").find("#testID")).toBeDefined();
  });

  it("sets a mediaQuery via props", () => {
    wrapper.setProps({ mediaQuery: "@media(max-width:768px){padding:8px}" });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("card__CardWrapper")).toHaveStyleRule(
      "padding",
      "8px",
      {
        media: "(max-width:768px)",
      }
    );
  });
});
