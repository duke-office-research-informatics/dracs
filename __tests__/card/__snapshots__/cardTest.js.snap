// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card adds a box-shadow when \`raised\` is set to true 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={true}
  height="auto"
  raised={true}
  width="auto"
>
  <section
    className="c0"
    height="auto"
    width="auto"
  />
</card__CardWrapper>
`;

exports[`Card attaches a className via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={true}
  className="testClass"
  height="auto"
  width="auto"
>
  <section
    className="c0 testClass"
    height="auto"
    width="auto"
  />
</card__CardWrapper>
`;

exports[`Card attaches children 1`] = `
.c1 {
  fill: #fff;
  vertical-align: top;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={true}
  height="auto"
  width="auto"
>
  <section
    className="c0"
    height="auto"
    width="auto"
  >
    <icon_wrapper__Icon
      aria-label="open customize and profile menu"
      color="#fff"
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
    >
      <svg
        aria-label="open customize and profile menu"
        className="c1"
        color="#fff"
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
      >
        <title>
          menu icon
        </title>
        <desc>
          hamburger menu icon
        </desc>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </icon_wrapper__Icon>
  </section>
</card__CardWrapper>
`;

exports[`Card does not apply a border when \`border\` set to false 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: auto;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={false}
  height="auto"
  width="auto"
>
  <section
    className="c0"
    height="auto"
    width="auto"
  />
</card__CardWrapper>
`;

exports[`Card matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={true}
  height="auto"
  width="auto"
>
  <section
    className="c0"
    height="auto"
    width="auto"
  />
</card__CardWrapper>
`;

exports[`Card sets a mediaQuery via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

@media (max-width:768px) {
  .c0 {
    padding: 8px;
  }
}

<card__CardWrapper
  border={true}
  height="auto"
  mediaQuery="@media(max-width:768px){padding:8px}"
  width="auto"
>
  <section
    className="c0"
    height="auto"
    width="auto"
  />
</card__CardWrapper>
`;

exports[`Card sets a style object via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={true}
  height="auto"
  style={
    Object {
      "flexDirection": "row",
    }
  }
  width="auto"
>
  <section
    className="c0"
    height="auto"
    style={
      Object {
        "flexDirection": "row",
      }
    }
    width="auto"
  />
</card__CardWrapper>
`;

exports[`Card sets a width via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: 100px;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={true}
  height="auto"
  width="100px"
>
  <section
    className="c0"
    height="auto"
    width="100px"
  />
</card__CardWrapper>
`;

exports[`Card sets an html id attr via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={true}
  height="auto"
  id="testID"
  width="auto"
>
  <section
    className="c0"
    height="auto"
    id="testID"
    width="auto"
  />
</card__CardWrapper>
`;

exports[`Card sets an onClick handler that is called on click 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
  cursor: pointer;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={true}
  height="auto"
  onClick={[Function]}
  width="auto"
>
  <section
    className="c0"
    height="auto"
    onClick={[Function]}
    width="auto"
  />
</card__CardWrapper>
`;

exports[`Card sets background color via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #f6f5f4;
  height: auto;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  bgColor="#f6f5f4"
  border={true}
  height="auto"
  width="auto"
>
  <section
    className="c0"
    height="auto"
    width="auto"
  />
</card__CardWrapper>
`;

exports[`Card sets height via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: 300px;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={true}
  height="300px"
  width="auto"
>
  <section
    className="c0"
    height="300px"
    width="auto"
  />
</card__CardWrapper>
`;

exports[`Card sets margin via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
  margin: 20px;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={true}
  height="auto"
  margin="20px"
  width="auto"
>
  <section
    className="c0"
    height="auto"
    width="auto"
  />
</card__CardWrapper>
`;

exports[`Card sets minHeight via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
  min-height: 200px;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={true}
  height="auto"
  minHeight="200px"
  width="auto"
>
  <section
    className="c0"
    height="auto"
    width="auto"
  />
</card__CardWrapper>
`;

exports[`Card sets minWidth via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
  min-width: 200px;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={true}
  height="auto"
  minWidth="200px"
  width="auto"
>
  <section
    className="c0"
    height="auto"
    width="auto"
  />
</card__CardWrapper>
`;

exports[`Card sets padding via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  height: auto;
  width: auto;
  border: solid 1px #e8e5e2;
  border-radius: 4px;
  overflow: hidden;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -ms-overflow-style: none;
  padding: 8px;
}

.react-draggable-dragging .c0 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  z-index: 100;
}

<card__CardWrapper
  border={true}
  height="auto"
  padding="8px"
  width="auto"
>
  <section
    className="c0"
    height="auto"
    width="auto"
  />
</card__CardWrapper>
`;
